src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+    private static final DateConvertor DATE_CONVERTOR = new DateConvertor();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate fromDate = DATE_CONVERTOR.convert(dateFrom);
+        LocalDate toDate = DATE_CONVERTOR.convert(dateTo);
+        StringBuilder salaryList = new StringBuilder();
+        salaryList.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (String name : names) {
+            int salary = 0;
+            for (String elementOfData : data) {
+                String[] dataItem = elementOfData.split(" ");
+                final String employeeName = dataItem[1];
+                final int hoursWorked = Integer.parseInt(dataItem[2]);
+                final int ratePerHour = Integer.parseInt(dataItem[3]);
+                LocalDate dateElement = DATE_CONVERTOR.convert(dataItem[0]);
```

Comment:
[Any magic numbers should be constants](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#any-magic-numbers-should-be-constants)
we can introduce constants like `HOURS_WORKED_INDEX` or something else